### DISCUSSION TITLE
Internal http requests make server hang

### DISCUSSION THREAD

**Discussion by randoms** [open-webui/open-webui#4496] [General]
# Bug Report

## Installation Method

git clone

## Environment

- **Open WebUI Version:** master branch
- **Ollama (if applicable):** 0.3.0

- **Operating System:** Ubuntu 20.04
- **Browser (if applicable):** Firefox

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

the server should response in time in any circumstance

## Actual Behavior:

server no response to all requests

## Description

**Bug Summary:**
There are many internal http requests in server code, for example request tts audio data from openai, the requests are executed in the main thread, if openai server responses slow, or openai server being blocked. The whole server will not response to any new requests. I suggest to make these requests in a separate thread. Just like the code below.

```python
r = await asyncio.to_thread(requests.post,
    url=f"{app.state.config.TTS_OPENAI_API_BASE_URL}/audio/speech",
    data=body,
    headers=headers,
    stream=True
)
```

**Comment by tjbck** [comment]
Feel free to make a PR after thorough testing!
